ballerina/http_connection.bal

```diff
@@ -292,7 +291,28 @@ isolated function arrayToString(string[] arr) returns string {
     return arrayString.substring(1, arrayString.length() - 1);
 }
 
-isolated function setPayload(anydata payload, Response response, string? mediaType = (), boolean setETag = false) {
+isolated function addLinksToPayload(anydata message, map<Link>? links) returns [boolean, anydata] {
+    if links !is () {
```

Comment:
```suggestion
    if links is map<Link> {
```

Reply:
I am creating and returning this map from the native side (HttpCallableUnitCallback). And when we check this as `if links is map<Link>` it returns false. Not sure why

Reply:
Does it work for `map<json>`?

Reply:
No, not worked for `map<json>` or `json`. There was a bug when creating the `BMap` and fixed via [7643324](https://github.com/ballerina-platform/module-ballerina-http/pull/1167/commits/7643324a218e8f51fd493139840d9752a06a7555)